### PR TITLE
fix: Remove 3PI config url validation

### PR DESCRIPTION
**Pull request by BigTailWolf**
Remove the url validation for TPC support.
Failed configuration will be on user's own risk in future.

### CODE REVIEW

**Comment by clundin25** [comment]
nit: Using the `$ git revert $COMMIT` could be an easier way to revert this code. This will also result in self-documenting commits.

**Comment by BigTailWolf** [comment]
> 

I considered that. But I found those validations are not added the same time. And very hard to find all these separate changes.

**Comment by BigTailWolf** [comment]
> nit: Using the `$ git revert $COMMIT` could be an easier way to revert this code. This will also result in self-documenting commits.

I considered that. But I found those validations are not added the same time. And very hard to find all these separate changes.

**Comment by BigTailWolf** [comment]
> Please update the README to mention that we do not validate the URLs in the config for Workload/Workforce

Hi @lsirac , and for the `is_workforce_pool`, we also have regex matching there, that would fail TPC user's check.

**Comment by lsirac** [comment]
> 

Yes, please update the regex as well

**Comment by BigTailWolf** [comment]
> > > Please update the README to mention that we do not validate the URLs in the config for Workload/Workforce
> > 
> > 
> > Hi @lsirac , and for the `is_workforce_pool`, we also have regex matching there, that would fail TPC user's check.
> 
> ~Yes, please update the regex as well~
> 
> Let's keep it for now, and circle back on this

Good. Maybe need backlog a "loose the pattern check for is_workforce_pool"

**Review by lsirac — DISMISSED** [review]

File: docs/user-guide.rst [hunk 1]
```diff
@@ -823,6 +823,16 @@ in the target project. Using `ImpersonatedCredentials` will allow the source_cre
 to assume the identity of a target_principal that does have access.
 
 
+Security considerations on configuration External and Impersonated credentials
```
Comment (lsirac) [inline docs/user-guide.rst:826]: lets add this in the workload and workforce section like Alex did

**Comment by BigTailWolf** [comment]
@arithmetic1728 may I get an approval?

File: docs/user-guide.rst [hunk 2]
```diff
@@ -793,6 +803,13 @@ Cloud resources from an OIDC or SAML provider.
     https://cloud.google.com/iam/docs/workforce-identity-federation#workforce-pools-user-project
 
 
+Note that this library does not perform any validation on the token_url,
```
Comment (lsirac) [inline docs/user-guide.rst:806]: Missing Security considerations header?

Comment (BigTailWolf) [reply docs/user-guide.rst:806]: No, this section already have header a few lines above

Comment (clundin25) [inline docs/user-guide.rst:806]: Why duplicate this snippet?

Comment (BigTailWolf) [reply docs/user-guide.rst:806]: To have this in both workload and workforce section since they are already separated.

**Review by clundin25 — APPROVED** [review]

**Pull request merged**